hal/network/ncp_client/quectel/quectel_ncp_client.cpp

```diff
@@ -1197,6 +1199,9 @@ int QuectelNcpClient::configureApn(const CellularNetworkConfig& conf) {
             const int r = CHECK_PARSER(resp.readResult());
             CHECK_TRUE(r == AtResponse::OK, SYSTEM_ERROR_UNKNOWN);
             netConf_ = networkConfigForImsi(buf, strlen(buf));
+            if (netConf_.hasNetProv()) {
+                netProv_ = netConf_.netProv();
+            }
```

Comment:
Even if `netConf_` is valid (which means `Cellular.setCredentials()` was called), we still need to make a call potentially to `networkConfigForImsi()` and copy the provider value out of it.

Same with SARA code.

Reply:
It defaults to CellularNetworkProv::NONE, and we don't support 3rd Party SIMs.  So if someone is using our SIM externally, they just need to ensure that credentials are cleared such that Device OS does the right thing.  If someone is using a SIM externally that matches one we use, they might not like settings we apply.  Configuring things properly will be up to them in the user app.

Reply:
I think this is arguable. If we can correctly detect that we are on a particular operator SIM even if we are using an external SIM with potentially custom apn/username/password set, we should still perform the lookup based on the ICCID and understand which one it is. If it's an unknown 3rdparty SIM, that's fine, we'll default to NONE, but otherwise if we have some workarounds for a particular operator in place they should work no matter what.

Moreover, we don't have an API to set the operator from the application.